### PR TITLE
fix: do not close connections when sigserver goes away

### PR DESCRIPTION
**Pull request by achingbrain**
Tweaks the logic around server errors so that if we lose our connection to the sigserver, we now let socket.io's reconnection logic handle trying to reconnect instead of emitting an error that causes everything to be torn down.

Fixes: #453

### CODE REVIEW

**Comment by achingbrain** [comment]
> example use case

I could be misreading, but this sounds like it's solving a slightly different problem - e.g. that the list of sigservers is a bit more speculative - they could be up, they could be down, and we should try to connect in future if they appear.

It sounds like it could be done in a separate PR to this one?

**Comment by abetaev** [comment]
i was trying to understand the purpose of `previouslyConnected` when came to this suggestion.

why is it a specific case to handle first failed connection as transport failure?

**Comment by achingbrain** [comment]
> why is it a specific case to handle first failed connection as transport failure?

It's to give the user fast feedback that their configuration isn't correct.  A similar thing happens with, for example, the tcp transport trying to listen on a port that's in use - it'll throw on startup rather than waiting to see if the port becomes available at some point in the future.

Trying to connect to remote signalling servers as part of node startup may be a little different since they aren't necessarily under your control, we could add an option to allow tolerance for failures around this, for example.  We may not wish to have this enabled by default though as it can become hard to reason about the correctness of your config on startup otherwise.

File: packages/webrtc-star-transport/src/listener.ts [hunk 1]
```diff
@@ -49,27 +48,46 @@ class SigServer extends EventEmitter<SignalServerServerEvents> implements Signal
...
+      if (previouslyConnected) {
+        this.dispatchEvent(new CustomEvent('reconnect'))
+      }
+    })
     this.socket.once('connect', () => {
+      // make sure we can reconnect in future
+      previouslyConnected = true
       this.dispatchEvent(new CustomEvent('listening'))
     })
+    this.socket.on('disconnect', () => {
```
Comment (mpetrunic) [inline packages/webrtc-star-transport/src/listener.ts:88]: maybe out of scope for this PR but shouldn't we remove all event listeners on close?

**Review by mpetrunic — APPROVED** [review]

File: packages/webrtc-star-transport/src/listener.ts (hunk 1 above)
Comment (achingbrain) [reply packages/webrtc-star-transport/src/listener.ts:88]: We do on line 214

**Pull request merged**